akka-stream-tests/src/test/scala/akka/stream/scaladsl/pattern/FlowDebugSpec.scala

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (C) 2014-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl.pattern
+
+import akka.event.{ DummyClassForStringSources, Logging }
+import akka.stream.scaladsl.{ Flow, Sink, Source, TestConfig }
+import akka.testkit.TestProbe
+
+import scala.concurrent.forkjoin.ThreadLocalRandom.{ current ⇒ random }
+
+import akka.stream.{ FlowMaterializer, ActorFlowMaterializer, ActorFlowMaterializerSettings }
+import akka.stream.testkit.{ AkkaSpec, StreamTestKit }
+import akka.stream.testkit.ScriptedTest
+
+import scala.util.control.NoStackTrace
+
+class FlowDebugSpec extends AkkaSpec("akka.loglevel = DEBUG") with ScriptedTest {
+
+  val settings = ActorFlowMaterializerSettings(system)
+    .withInputBuffer(initialSize = 2, maxSize = 16)
+
+  implicit val mat: FlowMaterializer = ActorFlowMaterializer()
+
+  def logProbe(): TestProbe = {
+    val p = TestProbe()
+    system.eventStream.subscribe(p.ref, classOf[Logging.Debug])
+    system.eventStream.subscribe(p.ref, classOf[Logging.Info])
+    system.eventStream.subscribe(p.ref, classOf[Logging.Warning])
+    system.eventStream.subscribe(p.ref, classOf[Logging.Error])
```

Comment:
1) Oh, right - will do.
2) Ah you mean `EventFilter`, yes that would be cleaner - I'll use it.